{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "#オーバーラップしているか検索"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import time"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19618 47716 67334\n"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_seq(chromosome,start,end,strand,window):\n",
    "    test_data = open(\"/Users/miyakesho/Google_Drive/master/research/txt/{}.txt\".format(chromosome), \"r\")\n",
    "    sequence=test_data.read()\n",
    "    target_seq=sequence[start-window:end+window]\n",
    "    target_seq=bigletter(target_seq)\n",
    "    if(strand==\"-\"):\n",
    "        target_seq=convert(target_seq)\n",
    "    return target_seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [],
   "source": [
    "def bigletter(seq):\n",
    "    new_seq=\"\"\n",
    "    for i in seq:\n",
    "        if(i==\"a\"):\n",
    "            new_seq+=\"A\"\n",
    "        elif(i==\"g\"):\n",
    "            new_seq+=\"G\"\n",
    "        elif(i==\"c\"):\n",
    "            new_seq+=\"C\"\n",
    "        elif(i==\"t\"):\n",
    "            new_seq+=\"T\"\n",
    "        else:\n",
    "            new_seq+=i\n",
    "    return new_seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [],
   "source": [
    "def convert(sequence):\n",
    "    \n",
    "    i=len(sequence)-1\n",
    "    convert_seq=\"\"\n",
    "    while(i>=0):\n",
    "        letter=sequence[i]\n",
    "        if(letter==\"A\"):\n",
    "            convert_seq+=\"T\"\n",
    "        elif(letter==\"T\"):\n",
    "            convert_seq+=\"A\"\n",
    "        elif(letter==\"G\"):\n",
    "            convert_seq+=\"C\"\n",
    "        elif(letter==\"C\"):\n",
    "            convert_seq+=\"G\"\n",
    "        else:\n",
    "            print(\"ERROR\")\n",
    "        i-=1\n",
    "    \n",
    "   \n",
    "    return convert_seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [],
   "source": [
    "def for_miRNA(seq):\n",
    "    new_seq=\"\"\n",
    "    for i in seq:\n",
    "        if(i==\"A\"):\n",
    "            new_seq+=\"U\"\n",
    "        elif(i==\"T\"):\n",
    "            new_seq+=\"A\"\n",
    "        elif(i==\"G\"):\n",
    "            new_seq+=\"C\"\n",
    "        elif(i==\"C\"):\n",
    "            new_seq+=\"G\"\n",
    "    new_seq=new_seq[::-1]\n",
    "    return new_seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [],
   "source": [
    "def miRNA_combine(sequence,miRNA):\n",
    "    seed=miRNA[1:8]\n",
    "    match=[]\n",
    "    location=[]\n",
    "    for i in range(0,len(sequence)-len(seed)+1):\n",
    "        candidate=for_miRNA(sequence[i:i+7])\n",
    "        \n",
    "        if(seed==candidate):\n",
    "            print(sequence,candidate,seed)\n",
    "            location.append(round(i/len(sequence)))\n",
    "    \n",
    "    for j in location:\n",
    "        if(j==1):\n",
    "            match.append(\"downstream\")\n",
    "        else:\n",
    "            match.append(\"upstream\")\n",
    "    return match"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {},
   "outputs": [],
   "source": [
    "def calculate(ythdf1_list,miRNA):\n",
    "    then=time.time()\n",
    "    count_list=[] \n",
    "    window=0\n",
    "    while(window<=500):\n",
    "        count_dict={\"downstream\":0,\"upstream\":0}\n",
    "\n",
    "        for i in range(len(ythdf1_list)):\n",
    "            if(i%100==0):\n",
    "                now=time.time()\n",
    "                print(i,len(ythdf1_list),now-then)\n",
    "                \n",
    "            row=ythdf1_list.iloc[i]\n",
    "            chromosome=row.Chromosome.lower()\n",
    "            start=int(row.Start)\n",
    "            end=int(row.End)\n",
    "            strand=row.Strand\n",
    "            sequence=get_seq(chromosome,start,end,strand,window)\n",
    "\n",
    "            miRNA_bind_list=miRNA_combine(sequence,miRNA1)\n",
    "            if (len(miRNA_bind_list)!=0):\n",
    "                print(miRNA_bind_list)\n",
    "                for k in miRNA_bind_list:\n",
    "                    count_dict[k]+=1\n",
    "                print(chromosome,start,end,strand,sequence)\n",
    "\n",
    "        count=[window]\n",
    "        for i in count_dict:\n",
    "            count.append(count_dict[i])\n",
    "        count_list.append(count)\n",
    "        window+=25\n",
    "    count_list=pd.DataFrame(count_list,columns=[\"window\",\"upstream\",\"downstream\"])\n",
    "    count_list.to_csv(\"overlap_count.csv\",index=False)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {},
   "outputs": [],
   "source": [
    "def main():\n",
    "    \n",
    "    rep1_ythdf1_list=pd.read_csv(\"/Users/miyakesho/Google_Drive/master/research/csv/PARCLIP_1.csv\")\n",
    "    rep2_ythdf1_list=pd.read_csv(\"/Users/miyakesho/Google_Drive/master/research/csv/PARCLIP_2.csv\")\n",
    "    ythdf1_list=rep1_ythdf1_list.append(rep2_ythdf1_list)\n",
    "\n",
    "    miRNA155=\"UUAAUGCUAAUCGUGAUAGGGGUU\"\n",
    "    miRNA1=\"UGGAAUGUAAAGAAGUAUGUAU\"\n",
    "\n",
    "    calculate(ythdf1_list,miRNA1)\n",
    "    calculate(ythdf1,miRNA155)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 67334 5.7220458984375e-06\n",
      "100 67334 17.025189876556396\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-150-263240bbee7e>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mmain\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m<ipython-input-149-65bdd65d7519>\u001b[0m in \u001b[0;36mmain\u001b[0;34m()\u001b[0m\n\u001b[1;32m      8\u001b[0m     \u001b[0mmiRNA1\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"UGGAAUGUAAAGAAGUAUGUAU\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      9\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 10\u001b[0;31m     \u001b[0mcalculate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mythdf1_list\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mmiRNA1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     11\u001b[0m     \u001b[0mcalculate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mythdf1\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mmiRNA155\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     12\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-148-0e05963c7605>\u001b[0m in \u001b[0;36mcalculate\u001b[0;34m(ythdf1_list, miRNA)\u001b[0m\n\u001b[1;32m     16\u001b[0m             \u001b[0mend\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrow\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mEnd\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     17\u001b[0m             \u001b[0mstrand\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mrow\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mStrand\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 18\u001b[0;31m             \u001b[0msequence\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mget_seq\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mchromosome\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mstart\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mend\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mstrand\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mwindow\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     19\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     20\u001b[0m             \u001b[0mmiRNA_bind_list\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmiRNA_combine\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msequence\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mmiRNA1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-133-1e05c2fffe2a>\u001b[0m in \u001b[0;36mget_seq\u001b[0;34m(chromosome, start, end, strand, window)\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0mget_seq\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mchromosome\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mstart\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mend\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mstrand\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mwindow\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      2\u001b[0m     \u001b[0mtest_data\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"/Users/miyakesho/Google_Drive/master/research/txt/{}.txt\"\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mchromosome\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"r\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 3\u001b[0;31m     \u001b[0msequence\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtest_data\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      4\u001b[0m     \u001b[0mtarget_seq\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0msequence\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mstart\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0mwindow\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0mend\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mwindow\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m     \u001b[0mtarget_seq\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mbigletter\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtarget_seq\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/lib/python3.7/codecs.py\u001b[0m in \u001b[0;36mdecode\u001b[0;34m(self, input, final)\u001b[0m\n\u001b[1;32m    320\u001b[0m         \u001b[0;31m# decode input (taking the buffer into account)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    321\u001b[0m         \u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbuffer\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0minput\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 322\u001b[0;31m         \u001b[0;34m(\u001b[0m\u001b[0mresult\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mconsumed\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_buffer_decode\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0merrors\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfinal\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    323\u001b[0m         \u001b[0;31m# keep undecoded input until the next call\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    324\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbuffer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mconsumed\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "main()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3.1843370833333333"
      ]
     },
     "execution_count": 153,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "17.025*67334/100/60/60"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene_symbol</th>\n",
       "      <th>Name</th>\n",
       "      <th>Peak_sum</th>\n",
       "      <th>Mutation_sum</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Peak_mutation</th>\n",
       "      <th>Region</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin</td>\n",
       "      <td>45.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>CHR8</td>\n",
       "      <td>144991180.0</td>\n",
       "      <td>144991217.0</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin</td>\n",
       "      <td>45.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>CHR8</td>\n",
       "      <td>144992009.0</td>\n",
       "      <td>144992066.0</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin</td>\n",
       "      <td>45.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>CHR8</td>\n",
       "      <td>144995867.0</td>\n",
       "      <td>144995950.0</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin</td>\n",
       "      <td>45.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>CHR8</td>\n",
       "      <td>144991089.0</td>\n",
       "      <td>144991137.0</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin</td>\n",
       "      <td>45.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>CHR8</td>\n",
       "      <td>144995298.0</td>\n",
       "      <td>144995330.0</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin</td>\n",
       "      <td>45.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>CHR8</td>\n",
       "      <td>144994629.0</td>\n",
       "      <td>144994642.0</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin</td>\n",
       "      <td>45.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>CHR8</td>\n",
       "      <td>144993160.0</td>\n",
       "      <td>144993172.0</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin</td>\n",
       "      <td>45.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>CHR8</td>\n",
       "      <td>144989511.0</td>\n",
       "      <td>144989534.0</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3UTR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin</td>\n",
       "      <td>45.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>CHR8</td>\n",
       "      <td>144992684.0</td>\n",
       "      <td>144992698.0</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin</td>\n",
       "      <td>45.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>CHR8</td>\n",
       "      <td>145003884.0</td>\n",
       "      <td>145003925.0</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin</td>\n",
       "      <td>45.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>CHR8</td>\n",
       "      <td>145004125.0</td>\n",
       "      <td>145004147.0</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>CDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin</td>\n",
       "      <td>45.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>CHR8</td>\n",
       "      <td>144993365.0</td>\n",
       "      <td>144993394.0</td>\n",
       "      <td>-</td>\n",
       "      <td>7.0</td>\n",
       "      <td>CDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin</td>\n",
       "      <td>45.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>CHR8</td>\n",
       "      <td>144990829.0</td>\n",
       "      <td>144990867.0</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin</td>\n",
       "      <td>45.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>CHR8</td>\n",
       "      <td>144992513.0</td>\n",
       "      <td>144992535.0</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin</td>\n",
       "      <td>45.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>CHR8</td>\n",
       "      <td>144999767.0</td>\n",
       "      <td>144999798.0</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin</td>\n",
       "      <td>45.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>CHR8</td>\n",
       "      <td>144992653.0</td>\n",
       "      <td>144992678.0</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin</td>\n",
       "      <td>45.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>CHR8</td>\n",
       "      <td>144994936.0</td>\n",
       "      <td>144994953.0</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin</td>\n",
       "      <td>45.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>CHR8</td>\n",
       "      <td>144995196.0</td>\n",
       "      <td>144995217.0</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin</td>\n",
       "      <td>45.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>CHR8</td>\n",
       "      <td>144994349.0</td>\n",
       "      <td>144994378.0</td>\n",
       "      <td>-</td>\n",
       "      <td>4.0</td>\n",
       "      <td>CDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin</td>\n",
       "      <td>45.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>CHR8</td>\n",
       "      <td>144996095.0</td>\n",
       "      <td>144996126.0</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin</td>\n",
       "      <td>45.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>CHR8</td>\n",
       "      <td>144997158.0</td>\n",
       "      <td>144997210.0</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin</td>\n",
       "      <td>45.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>CHR8</td>\n",
       "      <td>144992847.0</td>\n",
       "      <td>144992874.0</td>\n",
       "      <td>-</td>\n",
       "      <td>89.0</td>\n",
       "      <td>CDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin</td>\n",
       "      <td>45.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>CHR8</td>\n",
       "      <td>144993949.0</td>\n",
       "      <td>144993968.0</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin</td>\n",
       "      <td>45.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>CHR8</td>\n",
       "      <td>144995718.0</td>\n",
       "      <td>144995761.0</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin</td>\n",
       "      <td>45.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>CHR8</td>\n",
       "      <td>144994706.0</td>\n",
       "      <td>144994730.0</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin</td>\n",
       "      <td>45.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>CHR8</td>\n",
       "      <td>144992581.0</td>\n",
       "      <td>144992604.0</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin</td>\n",
       "      <td>45.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>CHR8</td>\n",
       "      <td>144995799.0</td>\n",
       "      <td>144995824.0</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin</td>\n",
       "      <td>45.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>CHR8</td>\n",
       "      <td>144994480.0</td>\n",
       "      <td>144994522.0</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin</td>\n",
       "      <td>45.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>CHR8</td>\n",
       "      <td>144990520.0</td>\n",
       "      <td>144990538.0</td>\n",
       "      <td>-</td>\n",
       "      <td>10.0</td>\n",
       "      <td>CDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin</td>\n",
       "      <td>45.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>CHR8</td>\n",
       "      <td>145003308.0</td>\n",
       "      <td>145003327.0</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CDS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19588</th>\n",
       "      <td>PILRB</td>\n",
       "      <td>paired immunoglobin-like type 2 receptor beta</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CHR7</td>\n",
       "      <td>99947389.0</td>\n",
       "      <td>99947409.0</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>UTR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19589</th>\n",
       "      <td>SKP1P2</td>\n",
       "      <td>S-phase kinase-associated protein 1 pseudogene 2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CHR12</td>\n",
       "      <td>17142353.0</td>\n",
       "      <td>17142379.0</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>UTR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19590</th>\n",
       "      <td>DBF4B</td>\n",
       "      <td>DBF4 homolog B (S. cerevisiae)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CHR17</td>\n",
       "      <td>42826030.0</td>\n",
       "      <td>42826044.0</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>UTR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19591</th>\n",
       "      <td>ADAM1</td>\n",
       "      <td>ADAM metallopeptidase domain 1, pseudogene</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CHR12</td>\n",
       "      <td>112337919.0</td>\n",
       "      <td>112337939.0</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>UTR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19592</th>\n",
       "      <td>PLEKHA8P1</td>\n",
       "      <td>pleckstrin homology domain containing, family ...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>CHR12</td>\n",
       "      <td>45568295.0</td>\n",
       "      <td>45568316.0</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>UTR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19593</th>\n",
       "      <td>LOC100527964</td>\n",
       "      <td>uncharacterized LOC100527964</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CHR1</td>\n",
       "      <td>180239049.0</td>\n",
       "      <td>180239076.0</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>UTR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19594</th>\n",
       "      <td>LOC100505876</td>\n",
       "      <td>uncharacterized LOC100505876</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>CHR2</td>\n",
       "      <td>37429074.0</td>\n",
       "      <td>37429103.0</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>UTR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19595</th>\n",
       "      <td>LOC100507266</td>\n",
       "      <td>uncharacterized LOC100507266</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CHR4</td>\n",
       "      <td>4546559.0</td>\n",
       "      <td>4546577.0</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>UTR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19596</th>\n",
       "      <td>LOC100507032</td>\n",
       "      <td>uncharacterized LOC100507032</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>CHR3</td>\n",
       "      <td>129616859.0</td>\n",
       "      <td>129616878.0</td>\n",
       "      <td>+</td>\n",
       "      <td>4.0</td>\n",
       "      <td>Intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19597</th>\n",
       "      <td>HNRNPUL2-BSCL2</td>\n",
       "      <td>HNRNPUL2-BSCL2 readthrough</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CHR11</td>\n",
       "      <td>62493078.0</td>\n",
       "      <td>62493102.0</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19598</th>\n",
       "      <td>LOC100507246</td>\n",
       "      <td>uncharacterized LOC100507246</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>CHR17</td>\n",
       "      <td>74560591.0</td>\n",
       "      <td>74560615.0</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>UTR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19599</th>\n",
       "      <td>LOC100288637</td>\n",
       "      <td>OTU domain containing 7A pseudogene</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CHR15</td>\n",
       "      <td>31037383.0</td>\n",
       "      <td>31037400.0</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19600</th>\n",
       "      <td>LOC100506866</td>\n",
       "      <td>uncharacterized LOC100506866</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CHR2</td>\n",
       "      <td>179404232.0</td>\n",
       "      <td>179404244.0</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>UTR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19601</th>\n",
       "      <td>HOTAIRM1</td>\n",
       "      <td>HOXA transcript antisense RNA, myeloid-specifi...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CHR7</td>\n",
       "      <td>27139619.0</td>\n",
       "      <td>27139634.0</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>UTR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19602</th>\n",
       "      <td>LOC728558</td>\n",
       "      <td>uncharacterized LOC728558</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CHR10</td>\n",
       "      <td>97634078.0</td>\n",
       "      <td>97634105.0</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19603</th>\n",
       "      <td>LOC728431</td>\n",
       "      <td>uncharacterized LOC728431</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CHR1</td>\n",
       "      <td>37920804.0</td>\n",
       "      <td>37920826.0</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>UTR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19604</th>\n",
       "      <td>LOC286189</td>\n",
       "      <td>uncharacterized LOC286189</td>\n",
       "      <td>1.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>CHR8</td>\n",
       "      <td>69218658.0</td>\n",
       "      <td>69218678.0</td>\n",
       "      <td>-</td>\n",
       "      <td>16.0</td>\n",
       "      <td>Intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19605</th>\n",
       "      <td>LOC401320</td>\n",
       "      <td>uncharacterized LOC401320</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CHR7</td>\n",
       "      <td>30601104.0</td>\n",
       "      <td>30601125.0</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>UTR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19606</th>\n",
       "      <td>LOC100507250</td>\n",
       "      <td>uncharacterized LOC100507250</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CHR12</td>\n",
       "      <td>69069510.0</td>\n",
       "      <td>69069529.0</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>UTR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19607</th>\n",
       "      <td>LOC100505933</td>\n",
       "      <td>uncharacterized LOC100505933</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CHR10</td>\n",
       "      <td>111767026.0</td>\n",
       "      <td>111767042.0</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19608</th>\n",
       "      <td>LOC100630923</td>\n",
       "      <td>LOC100289561-PRKRIP1 readthrough</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CHR7</td>\n",
       "      <td>102036669.0</td>\n",
       "      <td>102036691.0</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19609</th>\n",
       "      <td>MIR548AE2</td>\n",
       "      <td>microRNA 548ae-2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CHR16</td>\n",
       "      <td>48376034.0</td>\n",
       "      <td>48376048.0</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19610</th>\n",
       "      <td>LOC284801</td>\n",
       "      <td>uncharacterized LOC284801</td>\n",
       "      <td>1.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>CHR20</td>\n",
       "      <td>26188742.0</td>\n",
       "      <td>26188762.0</td>\n",
       "      <td>-</td>\n",
       "      <td>14.0</td>\n",
       "      <td>Intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19611</th>\n",
       "      <td>HIST1H2APS1</td>\n",
       "      <td>histone cluster 1, H2a, pseudogene 1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CHR6</td>\n",
       "      <td>25732887.0</td>\n",
       "      <td>25732909.0</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>UTR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19612</th>\n",
       "      <td>LOC100652739</td>\n",
       "      <td>uncharacterized LOC100652739</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CHR6</td>\n",
       "      <td>150226715.0</td>\n",
       "      <td>150226730.0</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19613</th>\n",
       "      <td>LOC100507373</td>\n",
       "      <td>uncharacterized LOC100507373</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CHR19</td>\n",
       "      <td>14269009.0</td>\n",
       "      <td>14269028.0</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19614</th>\n",
       "      <td>LOC100508120</td>\n",
       "      <td>uncharacterized LOC100508120</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CHR6</td>\n",
       "      <td>2273065.0</td>\n",
       "      <td>2273076.0</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19615</th>\n",
       "      <td>LOC100506100</td>\n",
       "      <td>uncharacterized LOC100506100</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CHR9</td>\n",
       "      <td>131495322.0</td>\n",
       "      <td>131495332.0</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>UTR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19616</th>\n",
       "      <td>KIAA1656</td>\n",
       "      <td>KIAA1656 protein</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CHR22</td>\n",
       "      <td>30773860.0</td>\n",
       "      <td>30773879.0</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>UTR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19617</th>\n",
       "      <td>LINC00507</td>\n",
       "      <td>long intergenic non-protein coding RNA 507</td>\n",
       "      <td>1.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>CHR12</td>\n",
       "      <td>128407388.0</td>\n",
       "      <td>128407397.0</td>\n",
       "      <td>+</td>\n",
       "      <td>10.0</td>\n",
       "      <td>Intron</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>19618 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Gene_symbol                                               Name  \\\n",
       "0                PLEC                                            plectin   \n",
       "1                PLEC                                            plectin   \n",
       "2                PLEC                                            plectin   \n",
       "3                PLEC                                            plectin   \n",
       "4                PLEC                                            plectin   \n",
       "5                PLEC                                            plectin   \n",
       "6                PLEC                                            plectin   \n",
       "7                PLEC                                            plectin   \n",
       "8                PLEC                                            plectin   \n",
       "9                PLEC                                            plectin   \n",
       "10               PLEC                                            plectin   \n",
       "11               PLEC                                            plectin   \n",
       "12               PLEC                                            plectin   \n",
       "13               PLEC                                            plectin   \n",
       "14               PLEC                                            plectin   \n",
       "15               PLEC                                            plectin   \n",
       "16               PLEC                                            plectin   \n",
       "17               PLEC                                            plectin   \n",
       "18               PLEC                                            plectin   \n",
       "19               PLEC                                            plectin   \n",
       "20               PLEC                                            plectin   \n",
       "21               PLEC                                            plectin   \n",
       "22               PLEC                                            plectin   \n",
       "23               PLEC                                            plectin   \n",
       "24               PLEC                                            plectin   \n",
       "25               PLEC                                            plectin   \n",
       "26               PLEC                                            plectin   \n",
       "27               PLEC                                            plectin   \n",
       "28               PLEC                                            plectin   \n",
       "29               PLEC                                            plectin   \n",
       "...               ...                                                ...   \n",
       "19588           PILRB      paired immunoglobin-like type 2 receptor beta   \n",
       "19589          SKP1P2   S-phase kinase-associated protein 1 pseudogene 2   \n",
       "19590           DBF4B                     DBF4 homolog B (S. cerevisiae)   \n",
       "19591           ADAM1         ADAM metallopeptidase domain 1, pseudogene   \n",
       "19592       PLEKHA8P1  pleckstrin homology domain containing, family ...   \n",
       "19593    LOC100527964                       uncharacterized LOC100527964   \n",
       "19594    LOC100505876                       uncharacterized LOC100505876   \n",
       "19595    LOC100507266                       uncharacterized LOC100507266   \n",
       "19596    LOC100507032                       uncharacterized LOC100507032   \n",
       "19597  HNRNPUL2-BSCL2                         HNRNPUL2-BSCL2 readthrough   \n",
       "19598    LOC100507246                       uncharacterized LOC100507246   \n",
       "19599    LOC100288637                OTU domain containing 7A pseudogene   \n",
       "19600    LOC100506866                       uncharacterized LOC100506866   \n",
       "19601        HOTAIRM1  HOXA transcript antisense RNA, myeloid-specifi...   \n",
       "19602       LOC728558                          uncharacterized LOC728558   \n",
       "19603       LOC728431                          uncharacterized LOC728431   \n",
       "19604       LOC286189                          uncharacterized LOC286189   \n",
       "19605       LOC401320                          uncharacterized LOC401320   \n",
       "19606    LOC100507250                       uncharacterized LOC100507250   \n",
       "19607    LOC100505933                       uncharacterized LOC100505933   \n",
       "19608    LOC100630923                   LOC100289561-PRKRIP1 readthrough   \n",
       "19609       MIR548AE2                                   microRNA 548ae-2   \n",
       "19610       LOC284801                          uncharacterized LOC284801   \n",
       "19611     HIST1H2APS1               histone cluster 1, H2a, pseudogene 1   \n",
       "19612    LOC100652739                       uncharacterized LOC100652739   \n",
       "19613    LOC100507373                       uncharacterized LOC100507373   \n",
       "19614    LOC100508120                       uncharacterized LOC100508120   \n",
       "19615    LOC100506100                       uncharacterized LOC100506100   \n",
       "19616        KIAA1656                                   KIAA1656 protein   \n",
       "19617       LINC00507         long intergenic non-protein coding RNA 507   \n",
       "\n",
       "       Peak_sum  Mutation_sum Chromosome        Start          End Strand  \\\n",
       "0          45.0         169.0       CHR8  144991180.0  144991217.0      -   \n",
       "1          45.0         169.0       CHR8  144992009.0  144992066.0      -   \n",
       "2          45.0         169.0       CHR8  144995867.0  144995950.0      -   \n",
       "3          45.0         169.0       CHR8  144991089.0  144991137.0      -   \n",
       "4          45.0         169.0       CHR8  144995298.0  144995330.0      -   \n",
       "5          45.0         169.0       CHR8  144994629.0  144994642.0      -   \n",
       "6          45.0         169.0       CHR8  144993160.0  144993172.0      -   \n",
       "7          45.0         169.0       CHR8  144989511.0  144989534.0      -   \n",
       "8          45.0         169.0       CHR8  144992684.0  144992698.0      -   \n",
       "9          45.0         169.0       CHR8  145003884.0  145003925.0      -   \n",
       "10         45.0         169.0       CHR8  145004125.0  145004147.0      -   \n",
       "11         45.0         169.0       CHR8  144993365.0  144993394.0      -   \n",
       "12         45.0         169.0       CHR8  144990829.0  144990867.0      -   \n",
       "13         45.0         169.0       CHR8  144992513.0  144992535.0      -   \n",
       "14         45.0         169.0       CHR8  144999767.0  144999798.0      -   \n",
       "15         45.0         169.0       CHR8  144992653.0  144992678.0      -   \n",
       "16         45.0         169.0       CHR8  144994936.0  144994953.0      -   \n",
       "17         45.0         169.0       CHR8  144995196.0  144995217.0      -   \n",
       "18         45.0         169.0       CHR8  144994349.0  144994378.0      -   \n",
       "19         45.0         169.0       CHR8  144996095.0  144996126.0      -   \n",
       "20         45.0         169.0       CHR8  144997158.0  144997210.0      -   \n",
       "21         45.0         169.0       CHR8  144992847.0  144992874.0      -   \n",
       "22         45.0         169.0       CHR8  144993949.0  144993968.0      -   \n",
       "23         45.0         169.0       CHR8  144995718.0  144995761.0      -   \n",
       "24         45.0         169.0       CHR8  144994706.0  144994730.0      -   \n",
       "25         45.0         169.0       CHR8  144992581.0  144992604.0      -   \n",
       "26         45.0         169.0       CHR8  144995799.0  144995824.0      -   \n",
       "27         45.0         169.0       CHR8  144994480.0  144994522.0      -   \n",
       "28         45.0         169.0       CHR8  144990520.0  144990538.0      -   \n",
       "29         45.0         169.0       CHR8  145003308.0  145003327.0      -   \n",
       "...         ...           ...        ...          ...          ...    ...   \n",
       "19588       1.0           1.0       CHR7   99947389.0   99947409.0      +   \n",
       "19589       1.0           2.0      CHR12   17142353.0   17142379.0      +   \n",
       "19590       1.0           1.0      CHR17   42826030.0   42826044.0      +   \n",
       "19591       1.0           1.0      CHR12  112337919.0  112337939.0      +   \n",
       "19592       1.0           3.0      CHR12   45568295.0   45568316.0      -   \n",
       "19593       1.0           1.0       CHR1  180239049.0  180239076.0      -   \n",
       "19594       1.0           5.0       CHR2   37429074.0   37429103.0      +   \n",
       "19595       1.0           1.0       CHR4    4546559.0    4546577.0      +   \n",
       "19596       1.0           4.0       CHR3  129616859.0  129616878.0      +   \n",
       "19597       1.0           1.0      CHR11   62493078.0   62493102.0      -   \n",
       "19598       1.0           3.0      CHR17   74560591.0   74560615.0      +   \n",
       "19599       1.0           1.0      CHR15   31037383.0   31037400.0      +   \n",
       "19600       1.0           2.0       CHR2  179404232.0  179404244.0      +   \n",
       "19601       1.0           2.0       CHR7   27139619.0   27139634.0      +   \n",
       "19602       1.0           1.0      CHR10   97634078.0   97634105.0      -   \n",
       "19603       1.0           1.0       CHR1   37920804.0   37920826.0      -   \n",
       "19604       1.0          16.0       CHR8   69218658.0   69218678.0      -   \n",
       "19605       1.0           1.0       CHR7   30601104.0   30601125.0      -   \n",
       "19606       1.0           1.0      CHR12   69069510.0   69069529.0      -   \n",
       "19607       1.0           1.0      CHR10  111767026.0  111767042.0      -   \n",
       "19608       1.0           2.0       CHR7  102036669.0  102036691.0      +   \n",
       "19609       1.0           1.0      CHR16   48376034.0   48376048.0      +   \n",
       "19610       1.0          14.0      CHR20   26188742.0   26188762.0      -   \n",
       "19611       1.0           1.0       CHR6   25732887.0   25732909.0      +   \n",
       "19612       1.0           1.0       CHR6  150226715.0  150226730.0      +   \n",
       "19613       1.0           1.0      CHR19   14269009.0   14269028.0      +   \n",
       "19614       1.0           1.0       CHR6    2273065.0    2273076.0      +   \n",
       "19615       1.0           1.0       CHR9  131495322.0  131495332.0      +   \n",
       "19616       1.0           1.0      CHR22   30773860.0   30773879.0      -   \n",
       "19617       1.0          10.0      CHR12  128407388.0  128407397.0      +   \n",
       "\n",
       "       Peak_mutation  Region  \n",
       "0                2.0     CDS  \n",
       "1                1.0     CDS  \n",
       "2                2.0     CDS  \n",
       "3                2.0     CDS  \n",
       "4                1.0     CDS  \n",
       "5                2.0     CDS  \n",
       "6                1.0     CDS  \n",
       "7                1.0    3UTR  \n",
       "8                2.0     CDS  \n",
       "9                2.0     CDS  \n",
       "10               3.0     CDS  \n",
       "11               7.0     CDS  \n",
       "12               1.0     CDS  \n",
       "13               1.0     CDS  \n",
       "14               1.0     CDS  \n",
       "15               2.0     CDS  \n",
       "16               1.0     CDS  \n",
       "17               1.0     CDS  \n",
       "18               4.0     CDS  \n",
       "19               1.0     CDS  \n",
       "20               1.0     CDS  \n",
       "21              89.0     CDS  \n",
       "22               1.0     CDS  \n",
       "23               2.0     CDS  \n",
       "24               1.0     CDS  \n",
       "25               1.0     CDS  \n",
       "26               1.0     CDS  \n",
       "27               2.0     CDS  \n",
       "28              10.0     CDS  \n",
       "29               1.0     CDS  \n",
       "...              ...     ...  \n",
       "19588            1.0     UTR  \n",
       "19589            2.0     UTR  \n",
       "19590            1.0     UTR  \n",
       "19591            1.0     UTR  \n",
       "19592            3.0     UTR  \n",
       "19593            1.0     UTR  \n",
       "19594            5.0     UTR  \n",
       "19595            1.0     UTR  \n",
       "19596            4.0  Intron  \n",
       "19597            1.0  Intron  \n",
       "19598            3.0     UTR  \n",
       "19599            1.0  Intron  \n",
       "19600            2.0     UTR  \n",
       "19601            2.0     UTR  \n",
       "19602            1.0  Intron  \n",
       "19603            1.0     UTR  \n",
       "19604           16.0  Intron  \n",
       "19605            1.0     UTR  \n",
       "19606            1.0     UTR  \n",
       "19607            1.0  Intron  \n",
       "19608            2.0  Intron  \n",
       "19609            1.0  Intron  \n",
       "19610           14.0  Intron  \n",
       "19611            1.0     UTR  \n",
       "19612            1.0  Intron  \n",
       "19613            1.0  Intron  \n",
       "19614            1.0  Intron  \n",
       "19615            1.0     UTR  \n",
       "19616            1.0     UTR  \n",
       "19617           10.0  Intron  \n",
       "\n",
       "[19618 rows x 10 columns]"
      ]
     },
     "execution_count": 154,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rep1_ythdf1_list=pd.read_csv(\"/Users/miyakesho/Google_Drive/master/research/csv/PARCLIP_1.csv\")\n",
    "rep1_ythdf1_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.41114091709279904"
      ]
     },
     "execution_count": 156,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(rep1_ythdf1_list)/len(rep2_ythdf1_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
